src/hooks/useGasPrice.ts

```diff
@@ -59,31 +81,42 @@ const useGasPrice = (): {
   const [counter] = useIntervalCounter(REFRESH_DELAY)
   const provider = useWeb3ReadOnly()
 
-  const [gasPrice, gasPriceError, gasPriceLoading] = useAsync<BigNumber | undefined>(() => {
-    if (gasPriceConfigs) {
-      return getGasPrice(gasPriceConfigs)
-    }
-  }, [gasPriceConfigs, counter])
-
-  const [feeData, feeDataError, feeDataLoading] = useAsync<FeeData>(() => {
-    return provider?.getFeeData()
-  }, [provider, counter])
-
-  // Save the previous gas price so that we don't return undefined each time it's polled
-  const lastPrice = useRef<BigNumber>()
-  // Fallback to feeData.gasPrice for non-EIP-1559 networks
-  lastPrice.current = gasPrice || feeData?.maxFeePerGas || feeData?.gasPrice || lastPrice.current
-
-  const lastPrioFee = useRef<BigNumber>()
-  // Fallback to 0 for non-EIP-1559 networks
-  lastPrioFee.current = feeData ? feeData.maxPriorityFeePerGas || BigNumber.from(0) : lastPrioFee.current
-
-  return {
-    maxFeePerGas: lastPrice.current,
-    maxPriorityFeePerGas: lastPrioFee.current,
-    gasPriceError: gasPriceError || feeDataError,
-    gasPriceLoading: gasPriceLoading || feeDataLoading,
-  }
+  // Fetch gas price from oracles or get a fixed value
+  const [gasPrice, gasPriceError, gasPriceLoading] = useAsync<BigNumber | undefined>(
+    () => {
+      if (gasPriceConfigs) {
+        return getGasPrice(gasPriceConfigs)
+      }
+    },
+    [gasPriceConfigs, counter],
+    false,
+  )
+
+  // Fetch the gas fees from the blockchain itself
+  const [feeData, feeDataError, feeDataLoading] = useAsync<FeeData>(
+    () => {
+      if (!chain || !provider) return
+      return _getFeeData(provider, hasFeature(chain, FEATURES.EIP1559))
+    },
+    [chain, provider, counter],
+    false,
+  )
+
+  // Prepare the return values
+  const maxFee = gasPrice || feeData?.maxFeePerGas || undefined
```

Comment:
If it's EIP-1559 capable, shouldn't we preferentially return `maxFeePerGas` first?

Reply:
No, `gasPrice` is the oracle price. `maxFeePerGas` from the chain is just a fallback.